File: backend/fingerprint_reader/fingerprints/models/gathering_model.py
from django.db import models
from django.core.validators import MinValueValidator, MaxValueValidator
from django.contrib.auth.models import User

from .baby_model import Baby


class Gathering(models.Model):
  fingerprint = models.ImageField()
  months = models.IntegerField(validators=[MinValueValidator(0), MaxValueValidator(12)])
  baby = models.ForeignKey(Baby, on_delete=models.PROTECT)
  user = models.ForeignKey(User, on_delete=models.PROTECT)

  created_at = models.DateTimeField(auto_now=True)
  updated_at = models.DateTimeField(auto_now_add=True)

  def __str__(self):
    return f'{self.baby.code} in {self.created_at.date}'

File: digital_reader/lib/models/gathering_model.dart
class GatheringModel {
  int? id;
  String? fingerprint;
  int? baby;
  int? months;
  String? createdAt;
  String? updatedAt;

  GatheringModel(
      {this.id,
      this.fingerprint,
      this.baby,
      this.months,
      this.createdAt,
      this.updatedAt});

  GatheringModel.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    fingerprint = json['fingerprint'];
    baby = json['baby'];
    months = json['months'];
    createdAt = json['created_at'];
    updatedAt = json['updated_at'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id'] = id;
    data['fingerprint'] = fingerprint;
    data['baby'] = baby;
    data['months'] = months;
    data['created_at'] = createdAt;
    data['updated_at'] = updatedAt;
    return data;
  }
}
